---
title: Compiler Error CS0156
ms.date: 07/20/2015
f1_keywords:
- CS0156
helpviewer_keywords:
- CS0156
ms.assetid: 32026b1b-bcd7-4464-b63f-3b38c00452a6
ms.openlocfilehash: 7ea5e861b71f5f65231bb29537a28a23a0821cef
ms.sourcegitcommit: 10986410e59ff29f2ec55c6759bde3eb4d1a00cb
ms.translationtype: MT
ms.contentlocale: cs-CZ
ms.lasthandoff: 05/31/2019
ms.locfileid: "66422972"
---
# <a name="compiler-error-cs0156"></a><span data-ttu-id="51f0f-102">Compiler Error CS0156</span><span class="sxs-lookup"><span data-stu-id="51f0f-102">Compiler Error CS0156</span></span>
<span data-ttu-id="51f0f-103">Příkaz throw bez argumentů není povolený v nakonec klauzuli, která je vnořená do nejbližší uzavírající klauzule catch</span><span class="sxs-lookup"><span data-stu-id="51f0f-103">A throw statement with no arguments is not allowed in a finally clause that is nested inside the nearest enclosing catch clause</span></span>  
  
 <span data-ttu-id="51f0f-104">A [throw](../../csharp/language-reference/keywords/throw.md) příkaz bez parametrů může být použit pouze v **catch** klauzuli, která nepřijímá žádné parametry.</span><span class="sxs-lookup"><span data-stu-id="51f0f-104">A [throw](../../csharp/language-reference/keywords/throw.md) statement with no parameters can only appear in a **catch** clause that takes no parameters.</span></span>  
  
 <span data-ttu-id="51f0f-105">Další informace najdete v tématu [výjimek a zpracování výjimek](../../csharp/programming-guide/exceptions/index.md).</span><span class="sxs-lookup"><span data-stu-id="51f0f-105">For more information, see [Exceptions and Exception Handling](../../csharp/programming-guide/exceptions/index.md).</span></span>  
  
 <span data-ttu-id="51f0f-106">Následující ukázka generuje CS0156:</span><span class="sxs-lookup"><span data-stu-id="51f0f-106">The following sample generates CS0156:</span></span>  
  
```csharp  
// CS0156.cs  
using System;  
  
namespace MyNamespace  
{  
   public class MyClass2 : Exception  
   {  
   }  
  
   public class MyClass  
   {  
      public static void Main()  
      {  
         try  
         {  
            throw;   // CS0156  
         }  
  
         catch(MyClass2)  
         {  
            throw;   // this throw is valid  
         }  
      }  
   }  
}  
```
